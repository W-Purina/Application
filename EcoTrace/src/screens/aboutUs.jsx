import React, { useState, useEffect } from 'react';
import { ScrollView, View, StyleSheet, Dimensions, Text } from 'react-native';

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;

const AboutUsScreen = ({ navigation }) => {


    return (
        <View style={styles.screen} contentContainerStyle={styles.contentContainer}>
            <View style={styles.greenHeader}>

            </View>

            <View style={styles.mainContainer}>
                <View style={styles.textcon}>
                    <Text style={styles.title}>EcoTrace: Your Partner in Green Travel</Text>
                    <Text style={styles.aboutustxt}>Innovative Technology, Sustainable Future{'\n\n'}

                        Founded by the visionary team "Keep Real" at the University of Auckland, EcoTrace is not just an app; it's a movement towards a greener tomorrow. Our diverse team of IT graduates, including experts in machine learning, financial analysis, and software development, is dedicated to making sustainable living a practical reality.{'\n\n'}

                        At EcoTrace, we understand the urgency of addressing the carbon footprint crisis. Our mission is to empower individuals and organisations to make informed, eco-friendly choices. With our cutting-edge app, we bring you a personalised carbon footprint tracking experience that's both user-friendly and impactful.{'\n\n'}

                        Our innovative app provides real-time data on transportation-related carbon emissions, encouraging users to choose greener travel options. Alongside, we offer engaging features like a map interface for easy tracking and a coupon system to reward sustainable choices.
                    </Text>
                    <Text style={styles.title}>Join Us on the Journey to Sustainability</Text>
                    <Text style={styles.aboutustxt}>Your choices matter. With EcoTrace, make each step count towards a healthier planet.{'\n\n'}
                        Contact us: ecotracenz@gmail.com
                    </Text>
                </View>

            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    screen: {
        width: screenWidth,
        height: screenHeight,
    },
    contentContainer: {
        paddingBottom: 20,
    },
    greenHeader: {
        width: screenWidth,
        height: screenHeight * 0.28,
        backgroundColor: '#B0D9B1',
    },
    mainContainer: {
        borderRadius: 20,
        width: screenWidth * 0.95,
        marginTop: -screenHeight * 0.27,
        height: screenHeight * 0.85,
        marginLeft: 10,
        marginRight: 20,
        backgroundColor: 'white',
        elevation: 3,
    },
    title: {
        marginTop: 10,
        fontSize: 18,
        fontWeight: '700',
        marginBottom: 25,
    },
    textcon: {
        marginHorizontal: 20,
        marginTop: 35,
        alignSelf: 'center',
        alignItems: 'center',
        justifyContent: 'center',
        margin: 5,
    },
    aboutustxt: {
        color: '#3E3838',
        fontSize: 13,
        fontWeight: '400',
        marginBottom: 15,
        //marginHorizontal: 15,
        lineHeight: 17,
    },
});

export default AboutUsScreen;