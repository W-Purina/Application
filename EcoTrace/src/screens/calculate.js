const calculateCarbonFootprint = async (distance, mode, make = null, type = null,) => {
  // Define carbon emission factors for different travel patterns
  const footprintPerMode = {
    walking: 0,
    bicycling: 0,
    bus: 0.153,
    electric: 0.012,
    rail: 0.019,
    taxi: 0.159,
    driving: 0.159,
  };

  // Get the carbon emission factor for the current model
  let coefficient;
  if (make && type) {
    coefficient = await fetchPredictionFromServer(make, type);
  } else {
    coefficient = footprintPerMode[mode.toLowerCase()] || 0;
  }


  // Calculate the carbon footprint
  const carbonFootprint = ((distance / 1000) * coefficient).toFixed(2);




  return parseFloat(carbonFootprint);
};


const fetchPredictionFromServer = async (make, vehicleType) => {
  try {
    const response = await fetch(
      //This 10.0.2.2 is the default ip address of the emulator on this computer. 
      //If it is a physical computer, this IP address should be changed
      `http://10.0.2.2:3000/predict?make=${encodeURIComponent(
        make,
      )}&vehicleType=${encodeURIComponent(vehicleType)}`,
      {
        method: 'POST',
      },
    );
    const text = await response.text();
    

    const data = JSON.parse(text);
    

    //Process and return the predicted value 
    const predictionRaw = data.prediction.trim();
    const predictionValue = parseFloat(predictionRaw);
    return predictionValue;
  } catch (error) {
    console.error('Error fetching prediction:', error);
  }
};


export { calculateCarbonFootprint, fetchPredictionFromServer };
