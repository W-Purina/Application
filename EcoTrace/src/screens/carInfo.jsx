import React, { useState, useEffect, useContext } from 'react';
import { Text, View, FlatList, StyleSheet, Modal, Image, TextInput, TouchableOpacity, Dimensions, Alert } from 'react-native';
import { useUser } from './userContext.js';
import { updateVehicleInfo, saveVehicleInfo, fetchSpecificVehicleInfo, deleteVehicleInfo } from './firestore.js';
import AppContext from '../theme/AppContext';

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;

const CarInfoScreen = ({ navigation, route }) => {

    const [selectedItem, setSelectedItem] = React.useState(null);
    const [query, setQuery] = React.useState('');
    const [filteredData, setFilteredData] = React.useState(null);
    const [currentDataSet, setCurrentDataSet] = useState(null);
    const [ds, setds] = React.useState(null);
    const { userId } = useUser();
    const [vehicleInfo, setVehicleInfo] = useState({});
    const [originalVehicleInfo, setOriginalVehicleInfo] = useState({});
    const vehicleId = route.params?.vehicleId;
    const { setShowOverlay } = useContext(AppContext);
    const data1 = [
        'Mercedes_Benz',
        'Porsche',
        'Toyota',
        'Audi',
        'Nissan',
        'Jeep',
        'Kia',
        'Honda',
        'Hyundai',
        'Volkswagen',
        'Mazda',
        'Lexus',
        'Subaru',
        'Volvo',
        'Mitsubishi',
        'Fiat',
        'Others',
    ];
    const data2 = [
        '2010',
        '2011',
        '2012',
        '2013',
        '2014',
        '2015',
        '2016',
        '2017',
        '2018',
        '2019',
        '2020',
        '2021',
        '2022',
        '2023',
        'others',
    ];
    const data3 = ['Petrol', 'Diesel', 'Electric', 'Hybrid'];
    const data4 = ['Suv', 'Hatchback', 'Sedan'];
    const [isModalVisible, setModalVisible] = useState(false);

    useEffect(() => {
        const loadVehicleData = async () => {
            if (vehicleId) {
                try {
                    const vehicleData = await fetchSpecificVehicleInfo(userId, vehicleId);
                    if (vehicleData) {
                        setVehicleInfo(vehicleData);
                        setOriginalVehicleInfo(vehicleData);
                    } else {
                        console.log('No vehicle data found for ID:', vehicleId);
                    }
                } catch (error) {
                    console.error('Error fetching vehicle info:', error);
                }
            }
        };

        loadVehicleData();
    }, [userId, vehicleId]);


    const toggleModal = () => {
        setModalVisible(!isModalVisible);
    };

    const handleSearch = text => {
        setQuery(text);
        if (text === '') {
            setFilteredData(currentDataSet.sort());
        } else {
            const filtered = currentDataSet.filter(item =>
                item.toLowerCase().includes(text.toLowerCase()),
            );
            setFilteredData(filtered.sort());
        }
    };

    const handleItemPress = item => {
        setSelectedItem(item);
        setQuery(item);
    };

    const handleConfirm = () => {
        if (!selectedItem) {
            Alert.alert("Error", "Please select an option", [
                { text: 'OK' }
            ]);
            return;
        }

        const updatedVehicleInfo = { ...vehicleInfo };
        switch (ds) {
            case 1:
                updatedVehicleInfo.carBrand = selectedItem;
                break;
            case 2:
                updatedVehicleInfo.carYear = selectedItem;
                break;
            case 3:
                updatedVehicleInfo.fuelType = selectedItem;
                break;
            case 4:
                updatedVehicleInfo.carType = selectedItem;
                break;
        }
        setQuery('');
        toggleModal();
        setVehicleInfo(updatedVehicleInfo);
    };

    const handleStore = async () => {
        setShowOverlay(true);
        try {
            if (vehicleId && JSON.stringify(vehicleInfo) === JSON.stringify(originalVehicleInfo)) {
                Alert.alert("Notice", "No changes were made to the vehicle information", [
                    {
                        text: 'OK', onPress: () => {
                            setShowOverlay(false);
                            navigation.goBack();
                        }
                    },
                ]);
                return;
            }

            if (vehicleId) {
                await updateVehicleInfo(userId, vehicleId, vehicleInfo);
                Alert.alert('Success', 'Vehicle information updated successfully', [
                    {
                        text: 'OK', onPress: () => {
                            setShowOverlay(false);
                            navigation.goBack();
                        }
                    },
                ]);
            } else {
                if (!vehicleId && (!vehicleInfo.carBrand || !vehicleInfo.carYear || !vehicleInfo.fuelType || !vehicleInfo.carType)) {
                    Alert.alert("Error", "Please complete all fields before submitting", [
                        {
                            text: 'OK', onPress: () => {
                                setShowOverlay(false);
                            }
                        },
                    ]);
                    return;
                }
                await saveVehicleInfo(userId, vehicleInfo);
                Alert.alert('Success', 'Vehicle information saved successfully', [
                    {
                        text: 'OK', onPress: () => {
                            setShowOverlay(false);
                            navigation.goBack();
                        }
                    },
                ]);
            }
        } catch (error) {
            Alert.alert("Error", "Failed to update or add vehicle information");
        }

    };

    const handleDelete = async () => {
        Alert.alert(
            "Delete Vehicle",
            "Are you sure you want to delete this vehicle?",
            [
                { text: "Cancel", style: "cancel" },
                {
                    text: "Delete",
                    onPress: async () => {
                        setShowOverlay(true);
                        try {
                            await deleteVehicleInfo(userId, vehicleId);
                            Alert.alert('Success', 'Vehicle information deleted successfully', [
                                {
                                    text: 'OK', onPress: () => {
                                        setShowOverlay(false);
                                        navigation.goBack();
                                    }
                                },
                            ]);
                        } catch (error) {
                            Alert.alert("Error", "Failed to delete vehicle information");
                        }
                    }
                }
            ]
        );
    };


    return (
        <View style={styles.screen} contentContainerStyle={styles.contentContainer}>
            <View style={styles.greenHeader}></View>
            <View style={styles.mainContainer}>

                <View>
                    <View style={styles.pickerContainer}>
                        <TouchableOpacity
                            onPress={() => {
                                toggleModal();
                                setFilteredData(data1);
                                setCurrentDataSet(data1);
                                setds(1);
                            }}>
                            <Text style={styles.buttonText}>
                                {vehicleInfo.carBrand ? `Your Car Brand is:  ${vehicleInfo.carBrand}` : 'Click to Select Car Brands'}
                            </Text>
                        </TouchableOpacity>
                    </View>

                    <View style={styles.pickerContainer}>
                        <TouchableOpacity
                            onPress={() => {
                                toggleModal();
                                setFilteredData(data2);
                                setCurrentDataSet(data2);
                                setds(2);
                            }}>
                            <Text style={styles.buttonText}>
                                {vehicleInfo.carYear
                                    ? `The Year is :  ${vehicleInfo.carYear}`
                                    : 'Click to Select Car Year'}
                            </Text>
                        </TouchableOpacity>
                    </View>

                    <View style={styles.pickerContainer}>
                        <TouchableOpacity
                            onPress={() => {
                                toggleModal();
                                setFilteredData(data3);
                                setCurrentDataSet(data3);
                                setds(3);
                            }}>
                            <Text style={styles.buttonText}>
                                {vehicleInfo.fuelType
                                    ? `The Fuel type is :  ${vehicleInfo.fuelType}`
                                    : 'Click to Select Fuel type'}
                            </Text>
                        </TouchableOpacity>
                    </View>

                    <View style={styles.pickerContainer}>
                        <TouchableOpacity
                            onPress={() => {
                                toggleModal();
                                setFilteredData(data4);
                                setCurrentDataSet(data4);
                                setds(4);
                            }}>
                            <Text style={styles.buttonText}>
                                {vehicleInfo.carType
                                    ? `The Car type is :  ${vehicleInfo.carType}`
                                    : 'Click to Select Car type'}
                            </Text>
                        </TouchableOpacity>
                    </View>
                </View>
                <Modal
                    animationType="slide"
                    transparent={true}
                    visible={isModalVisible}
                    onRequestClose={toggleModal}>
                    <View style={styles.modalBackground}>
                        <View style={styles.modalContainer}>
                            <TextInput
                                style={styles.input2}
                                placeholder="Search here ..."
                                value={query}
                                onChangeText={handleSearch}
                            />
                            <FlatList
                                data={filteredData}
                                keyExtractor={item => item}
                                renderItem={({ item }) => (
                                    <TouchableOpacity onPress={() => handleItemPress(item)}>
                                        <Text
                                            style={[
                                                styles.item,
                                                {
                                                    backgroundColor:
                                                        selectedItem === item ? 'lightgray' : 'white',
                                                },
                                            ]}>
                                            {item}
                                        </Text>
                                    </TouchableOpacity>
                                )}
                            />
                            <TouchableOpacity onPress={handleConfirm} style={styles.bto}>
                                <Text style={styles.buttonText2}>Confirm</Text>
                            </TouchableOpacity>

                            {/* Close modal button */}
                        </View>
                    </View>
                </Modal>
                <View style={styles.con3}>
                    <TouchableOpacity onPress={handleStore} style={styles.bto1}>
                        <Text style={styles.buttonText2}>Confirm</Text>
                    </TouchableOpacity>
                    {vehicleId && (
                        <TouchableOpacity onPress={handleDelete} style={styles.bto1}>
                            <Text style={styles.buttonText2}>Delete</Text>
                        </TouchableOpacity>
                    )}
                </View>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    screen: {
        width: screenWidth,
        height: screenHeight,
    },
    contentContainer: {
        paddingBottom: 20,
    },
    greenHeader: {
        width: screenWidth,
        height: screenHeight * 0.28,
        backgroundColor: '#B0D9B1',
    },
    mainContainer: {
        borderRadius: 20,
        width: screenWidth * 0.95,
        height: screenHeight * 0.85,
        marginTop: -screenHeight * 0.27,
        marginLeft: 10,
        marginRight: 20,
        backgroundColor: 'white',
        elevation: 3,
    },
    imageHeader: {
        width: 72,
        height: 72,
        marginTop: 20,
        marginBottom: -10,
        borderRadius: 72 / 2,
        overflow: 'hidden',
        alignSelf: 'center',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0)',
    },
    image: {
        width: '100%',
        height: '100%',
        resizeMode: 'cover',
    },
    input: {
        marginTop: 41,
        width: '90%',
        height: 50,
        borderColor: '#D1D5DB',
        borderWidth: 1.5,
        paddingHorizontal: 18,
        alignSelf: 'center',
        borderRadius: 5,
        fontSize: 14,
        color: '#555555',
    },
    pickerContainer: {
        marginTop: 30,
        height: 50,
        width: '90%',
        borderColor: '#D1D5DB',
        borderWidth: 1.5,
        alignSelf: 'center',
        justifyContent: 'center',
        borderRadius: 5,
    },
    buttonText: {
        paddingHorizontal: 18,
        color: '#555555',
        fontSize: 14,
    },
    modalContainer: {
        width: '100%',
        height: '100%',
        backgroundColor: 'white',
        borderRadius: 20,
        padding: 15,
    },
    modalBackground: {
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
        width: '100%',
        height: '100%',
    },
    input2: {
        fontSize: 16,
        height: 55,
        margin: 12,
        borderWidth: 1,
        padding: 10,
    },
    item: {
        marginLeft: 10,
        padding: 10,
        fontSize: 18,
        height: 44,
    },
    bto: {
        height: 50,
        width: '90%',
        backgroundColor: '#609966',
        alignSelf: 'center',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 5,
        marginBottom: 20,
    },
    bto1: {
        height: 45,
        width: '90%',
        backgroundColor: '#609966',
        alignSelf: 'center',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 5,
        margin: 8,
    },
    buttonText2: {
        fontWeight: '700',
        fontSize: 16,
        color: 'white',
    },
    con3: {
        marginTop: 40,
    },
});

export default CarInfoScreen;
