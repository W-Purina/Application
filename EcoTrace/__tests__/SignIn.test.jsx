import 'react-native';
import React from 'react';
import SignInScreen from '../src/screens/signin';
import { UserProvider } from '../src/screens/userContext';
import { NavigationContainer } from '@react-navigation/native';
import renderer from 'react-test-renderer';
import { render, fireEvent, act } from '@testing-library/react-native';

// (1) Snapshot Testing - ensure that changes to the UI are controlled and intentional
it('renders SignInScreen correctly', async () => {

  const mockNavigation = {
    navigate: jest.fn(),
  };

  let tree;
  await act(async () => {
    tree = renderer.create(
      <UserProvider>
        <NavigationContainer>
          <SignInScreen navigation={mockNavigation} />
        </NavigationContainer>
      </UserProvider>
    );
  });
  expect(tree.toJSON()).toMatchSnapshot()
});


// (2) Component interactive testing
describe('SignInScreen', () => {
  it('fills out the form and presses the sign-in button', async () => {
    // 创建一个模拟的navigate函数用于测试导航功能
    const mockNavigate = jest.fn();
    // 渲染SignInScreen组件
    const { getByPlaceholderText, getByText } = render(
      <UserProvider>
        <NavigationContainer>
          <SignInScreen navigation={{ navigate: mockNavigate }} />
        </NavigationContainer>
      </UserProvider>
    );

    // 查找邮箱和密码输入框, “SIGN IN”按钮
    const emailInput = getByPlaceholderText('Email');
    const passwordInput = getByPlaceholderText('Password');
    const signInButton = getByText('SIGN IN');

    // 使用act包裹异步操作
    await act(async () => {
      fireEvent.changeText(emailInput, 'test@example.com');
      fireEvent.changeText(passwordInput, 'password123');
      fireEvent.press(signInButton);
    });

    // expect(mockNavigate).toHaveBeenCalledWith('BottomTabs');

  });

  it('displays an error message for invalid email format', async () => {
    const { getByPlaceholderText, getByText } = render(
      <UserProvider>
        <NavigationContainer>
          <SignInScreen navigation={{ navigate: jest.fn() }} />
        </NavigationContainer>
      </UserProvider>
    );

    const emailInput = getByPlaceholderText('Email');
    await act(async () => {
      fireEvent.changeText(emailInput, 'invalidemail');
    });

    expect(getByText('Invalid email format')).toBeTruthy();
  });

  it('navigates to forgot password screen', async () => {
    const mockNavigate = jest.fn();
    const { getByText } = render(
      <UserProvider>
        <NavigationContainer>
          <SignInScreen navigation={{ navigate: mockNavigate }} />
        </NavigationContainer>
      </UserProvider>
    );

    const forgotPasswordButton = getByText('Forgot your password?');
    await act(async () => {
      fireEvent.press(forgotPasswordButton);
    });

    expect(mockNavigate).toHaveBeenCalledWith('Forgot Password');
  });

});