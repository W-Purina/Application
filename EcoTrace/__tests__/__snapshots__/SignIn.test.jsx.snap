// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignInScreen correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#B0D9B1",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "marginLeft": 20,
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
            "fontSize": 30,
            "fontWeight": "bold",
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      Welcome back!
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderTopLeftRadius": 15,
        "borderTopRightRadius": 15,
        "flex": 1,
        "marginTop": 50,
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <TextInput
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "borderColor": "#DDD",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          {
            "borderColor": "#DDD",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "marginBottom": 10,
            "marginTop": 10,
            "padding": 10,
          }
        }
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 10,
            "position": "absolute",
            "right": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "grey",
                "fontSize": 20,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ó°›‘
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "marginBottom": 15,
              "textDecorationLine": "underline",
            }
          }
        >
          Forgot your password?
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "#000",
                "borderRadius": 2,
                "borderWidth": 1,
                "height": 15,
                "justifyContent": "center",
                "marginRight": 10,
                "width": 15,
              },
              null,
            ]
          }
        />
        <Text
          style={
            {
              "color": "#1F2937",
            }
          }
        >
          Remember me
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#B0D9B1",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginTop": 20,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontWeight": "bold",
          }
        }
      >
        SIGN IN
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 8,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#E5E7EB",
            "flex": 1,
            "height": 1,
            "marginHorizontal": 8,
          }
        }
      />
      <Text
        style={
          {
            "color": "#9CA3AF",
            "fontSize": 14,
            "marginBottom": 15,
            "marginTop": 15,
            "paddingHorizontal": 8,
          }
        }
      >
        or
      </Text>
      <View
        style={
          {
            "backgroundColor": "#E5E7EB",
            "flex": 1,
            "height": 1,
            "marginHorizontal": 8,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginVertical": 15,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#BABABA",
              "borderRadius": 25,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginHorizontal": 30,
              "width": 50,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/theme/images/google-icon.png",
              }
            }
            style={
              {
                "height": 30,
                "resizeMode": "contain",
                "width": 30,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#BABABA",
              "borderRadius": 25,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginHorizontal": 30,
              "width": 50,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/theme/images/facebook-icon.png",
              }
            }
            style={
              {
                "height": 30,
                "resizeMode": "contain",
                "width": 30,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#BABABA",
              "borderRadius": 25,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginHorizontal": 30,
              "width": 50,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/theme/images/apple-icon.png",
              }
            }
            style={
              {
                "height": 30,
                "resizeMode": "contain",
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 80,
          },
          {
            "marginTop": 150,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#000",
          }
        }
      >
        Donâ€™t have an account?
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#5E8C61",
              "fontWeight": "bold",
              "marginLeft": 5,
              "textDecorationLine": "underline",
            }
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
